src/marks/raster.js

```diff
@@ -383,6 +373,42 @@ function segmentProject(x1, y1, x2, y2, x, y) {
   return a > 0 && b > 0 ? a / (a + b) : +(a > b);
 }
 
+function cross(xa, ya, xb, yb) {
+  return xa * yb - xb * ya;
+}
+
+function ray(j, X, Y) {
+  const n = X.length;
+  const xc = X.at(j - 2),
+    yc = Y.at(j - 2),
+    xa = X.at(j - 1),
+    ya = Y.at(j - 1),
+    xb = X[j],
+    yb = Y[j],
+    xd = X.at(j + 1 - n),
+    yd = Y.at(j + 1 - n);
+  const dxab = xa - xb;
+  const dyab = ya - yb;
+  const dxca = xc - xa;
+  const dyca = yc - ya;
+  const dxbd = xb - xd;
+  const dybd = yb - yd;
+  const hab = Math.hypot(dxab, dyab);
+  const hca = Math.hypot(dxca, dyca);
+  const hbd = Math.hypot(dxbd, dybd);
+  return (x, y) => {
+    const dxa = x - xa;
+    const dya = y - ya;
+    const dxb = x - xb;
+    const dyb = y - yb;
+    return (
+      cross(dxa, dya, dxb, dyb) >= 0 &&
+      cross(dxa, dya, dxab, dyab) * hca >= cross(dxa, dya, dxca, dyca) * hab &&
+      cross(dxb, dyb, dxab, dyab) * hbd >= cross(dxb, dyb, dxbd, dybd) * hab
```

Comment:
maybe compare `> -0.01` to avoid precision issues that would make a point not part of the polygon on its left and not part of the polygon on its right?